cvat/apps/engine/background.py

```diff
@@ -144,12 +146,14 @@ def get_timestamp(self, time_: datetime) -> str:
 class ReenqueueJobException(Exception):
     pass
 
-def cancel_and_reenqueue(rq_job: RQJob) -> None:
-    # Cancel the current job and then reenqueue it the main export function.
-    # The new attempt will be made after the last existing job.
-    # In the case the server is configured with ONE_RUNNING_JOB_IN_QUEUE_PER_USER
-    # we have to enqueue dependent jobs after canceling one.
-    rq_job.cancel(enqueue_dependents=settings.ONE_RUNNING_JOB_IN_QUEUE_PER_USER)
+def recreate(rq_job: RQJob, should_be_canceled: bool = True) -> None:
+    # Cancel and delete a job and then raise ReenqueueJobException
+    # to indicate that the job should be re-created
+    if should_be_canceled:
+        # In the case the server is configured with ONE_RUNNING_JOB_IN_QUEUE_PER_USER
+        # we have to enqueue dependent jobs after canceling one.
+        rq_job.cancel(enqueue_dependents=settings.ONE_RUNNING_JOB_IN_QUEUE_PER_USER)
+
     rq_job.delete()
     raise ReenqueueJobException()
```

Comment:
- Using exceptions to control execution flow is typically considered an antipattern 
- I can't find any places where this exception is actually used, in all cases it's being suppressed